sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionService.java

```diff
@@ -191,6 +192,70 @@ public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<?,
 		return QueryHelper.getResultList(em, cq, first, max);
 	}
 
+	public List<Action> getActionList(ActionCriteria actionCriteria, Integer first, Integer max, List<SortProperty> sortProperties) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> action = cq.from(getElementClass());
+
+		// Add filters
+		Predicate filter = null;
+		if (actionCriteria == null || !actionCriteria.hasContextCriteria()) {
+			filter = createUserFilter(cb, cq, action);
+		}
+
+		if (actionCriteria != null) {
+			Predicate criteriaFilter = buildCriteriaFilter(actionCriteria, cb, action);
+			filter = CriteriaBuilderHelper.and(cb, filter, criteriaFilter);
+		}
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+		if (sortProperties != null && sortProperties.size() > 0) {
+			List<Order> order = new ArrayList<>(sortProperties.size());
+			for (SortProperty sortProperty : sortProperties) {
+				Expression<?> expression;
+				switch (sortProperty.propertyName) {
+				case ActionDto.UUID:
```

Comment:
I think you can simplify all of these (except the title because of the lowercase addition) by writing all the cases underneath each other and then just writing `action.get(sortProperty.propertyName)` because they're direct properties of actions and aren't computed; see e.g. EventFacadeEjb.getIndexList for an example